tests/all/resin_root_part/resin_root_part_bbb

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+
+. /tmp/test-lib.sh
+CONF=/mnt/boot/resinOS_uEnv.txt
+
+# we expect this to be '2' on a fresh system. '3' is also good.
+current=$(mount | grep 'on / type' | sed -e 's/ .*//' -e 's/^.*\(.\)/\1/') # e.g. 2
+dev=$(mount | grep 'on / type' | sed -e 's/ .*//' -e 's/[0-9]$//') # e.g. /dev/mmcblk1p
+
+test ! -e "$CONF" && echo resin_root_part=$current > "$CONF"
+
+if [ "$current" != "2" ] && [ "$current" != "3" ]; then
+    log ERROR "unexpected root partition number"
+fi
+
+configured=$(sed -ne 's/^resin_root_part=//p' "$CONF")
+if [ "$current" != "$configured" ]; then
+    rm -f /mnt/boot/.test
+    log FAILED "\$current ($current) != \$configured ($configured)"
+fi
+
+if [ -e /mnt/boot/.test ]; then
+    rm /mnt/boot/.test
+    log PASSED "OK"
+else
+    test "$current" = "2" && other=3 || other=2
+
+    log INFO "copying root partition from $dev$current to $dev$other..."
+    dd "if=$dev$current" "of=$dev$other" bs=1M
+    sed -i -e "s/^resin_root_part=.*/resin_root_part=$other/" "$CONF"
+    touch /mnt/boot/.test
+    /sbin/reboot
```

Comment:
Yes, the script must be ran twice. First time it copies a partition and reboots. The second time it checks if that action was successful. We need to somehow keep track of whether it's ran the first or second time.